dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/linelisting/trackedentityinstance/TeiAnalyticsQueryService.java

```diff
@@ -65,18 +64,23 @@
      * against the underline data provider and return. The results found will be
      * returned encapsulated on a Grid object.
      *
-     * @param teiParams
+     * @param teiLineListingParams
      * @return the populated Grid object
      * @throws IllegalArgumentException if the given teiParams is null
      */
-    public Grid getTeiGrid( final TeiParams teiParams )
+    public Grid getTeiGrid( final TeiLineListingParams teiLineListingParams )
     {
-        notNull( teiParams, "The 'teiParams' must not be null" );
+        notNull( teiLineListingParams, "The 'teiParams' must not be null" );
 
-        final Query query = teiJdbcQuery.from( teiParams );
-        final SqlQueryResult result = (SqlQueryResult) queryExecutor.execute( query );
-        final List<GridHeader> headers = from( ((SqlQuery) query).getColumns() );
+        final SqlQuery query = (SqlQuery) teiJdbcQuery.from( teiLineListingParams );
+        final SqlQueryResult result = queryExecutor.execute( query );
+        final List<GridHeader> headers = from( query.getColumns() );
 
         return gridAdaptor.createGrid( headers, result.result() );
     }
+
+    public void validateRequest( TeiLineListingRequest request )
```

Comment:
In our last discussion, I think we decided to move it into a separate component.

Reply:
Ok, I'll extract it into a dedicated tei validation service